src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -10,19 +11,20 @@ namespace Microsoft.Diagnostics.Tracing
     /// <summary>
     /// TraceLoggingEvnetId is a class that manages assigning event IDs (small 64k numbers)
     /// to TraceLogging Style events (which don't have them).  Because TraceEvent uses EventIDs
-    /// so fundamentally this difficiency is very problematic.   
+    /// so fundamentally this deficiency is very problematic.   
     /// 
     /// Arguably this should have been done by the ETW system itself.  
     /// 
     /// You use it by calling TestForTraceLoggingEventAndFixupIfNeeded on eventRecords.  
     /// You also have to explicitly call 'Dispose' when you are done with this class.  
     /// </summary>
-    internal unsafe struct TraceLoggingEventId
+    internal unsafe struct TraceLoggingEventId : IDisposable
     {
         /// <summary>
         /// Checks to see if eventRecord has TraceLogging meta data associated with it (EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
         /// and if so updates EventHeader.Id to be an event ID unique to that provider/opcode/meta-data blob. 
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
:question: Were you encountering situations where this method was not already getting inlined? I would prefer to omit this change in light of requests like #153 unless there is a demonstrated problem with the current approach.